src/sst/core/statapi/statlghistogram.h

```diff
@@ -0,0 +1,382 @@
+    }
+
+    ~LogBinHistogramStatistic() {}
+
+    LogBinHistogramStatistic() :
+        Statistic<BinDataType>()
+    {} // For serialization ONLY
+
+    virtual const std::string& getStatTypeName() const override { return stat_type_; }
+
+    void serialize_order(SST::Core::Serialization::serializer& ser) override
+    {
+        SST::Statistics::Statistic<BinDataType>::serialize_order(ser);
+        SST_SER(m_minValue);
+        SST_SER(m_binWidth);
+        SST_SER(m_numBins);
+        SST_SER(m_OOBMinCount);
+        SST_SER(m_OOBMaxCount);
+        SST_SER(m_itemsBinnedCount);
+        SST_SER(m_totalSummed);
+        SST_SER(m_totalSummedSqr);
+        SST_SER(m_binsMap);
+        SST_SER(m_dumpBinsOnOutput);
+        SST_SER(m_includeOutOfBounds);
+        // SST_SER(m_Fields); // Rebuilt by stat output object
+    }
+
+protected:
+    /**
+        Adds a new value to the histogram. The correct bin is identified and then incremented. If no bin can be found
+        to hold the value then a new bin is created.
+    */
+    void addData_impl_Ntimes(uint64_t const N, BinDataType const value) override
+    {
+        // Check to see if the value is above or below the min/max values
+        if ( value < getBinsMinValue() ) {
+            m_OOBMinCount += N;
+            return;
+        }
+        if ( value > getBinsMaxValue() ) {
+            m_OOBMaxCount += N;
+            return;
+        }
+
+        // This value is to be binned...
+        // Add the "in limits" value to the total summation's
+        m_totalSummed += static_cast<BinDataType>(N) * value;
+        m_totalSummedSqr += static_cast<BinDataType>(N) * (value * value);
+
+        // Increment the Binned count (note this <= to the Statistics added Item Count)
+        ++m_itemsBinnedCount;
+
+        // Figure out what the starting bin is and find it in the map
+        // To support signed and unsigned values along with floating point types,
+        // the calculation to find the bin_start value must be done in floating point
+        // then converted to BinDataType
+        double const      log2_value = std::log2(static_cast<double>(value));
+        double const      bin_floor_value = std::floor(log2_value); // Find the floor of the value
+        BinDataType const bin_start = static_cast<BinDataType>(bin_floor_value);
+        //      printf("DEBUG: value = %d, junk1 = %f, calc2 = %f, calc3 = %f : bin_start = %d, item count = %ld, \n",
+        //      value, calc1, calc2, calc3, bin_start, getStatCollectionCount());
+
+        HistoMapItr_t bin_itr = m_binsMap.find(bin_start);
+
+        // Was the bin found?
+        if ( bin_itr == m_binsMap.end() ) {
+            // No, Create the bin and set a value of 1 to it
+            m_binsMap.insert(std::pair<BinDataType, CountType>(bin_start, static_cast<CountType>(N)));
+        }
+        else {
+            // Yes, Increment the specific bin's count
+            bin_itr->second += static_cast<CountType>(N);
+        }
```

Comment:
```c++
        HistoMapItr_t bin_itr = m_binsMap.find(bin_start);

        // Was the bin found?
        if ( bin_itr == m_binsMap.end() ) {
            // No, Create the bin and set a value of 1 to it
            m_binsMap.insert(std::pair<BinDataType, CountType>(bin_start, static_cast<CountType>(N)));
        }
        else {
            // Yes, Increment the specific bin's count
            bin_itr->second += static_cast<CountType>(N);
        }
```
... can be replaced with one line:

```c++
m_binsMap[bin_start] += static_cast<CountType>(N); // Increment the specific bin's count
```
The `std::map` `operator[]` creates a value-initialized new entry if one does not exist, and returns a reference to the entry. _Value-initialized_ means `0` for arithmetic types.